gecko-media/src/player.rs

```diff
@@ -524,24 +617,26 @@ fn to_ffi_vec(bytes: Vec<u8>) -> RustVecU8Object {
     }
 }
 
-fn to_ffi_planar_ycbycr_images(size: usize, elements: *mut GeckoPlanarYCbCrImage) -> Vec<PlanarYCbCrImage> {
+fn to_ffi_planar_ycbycr_images(
+    size: usize,
+    elements: *mut GeckoPlanarYCbCrImage,
+    video_frame_allocator: &Arc<Mutex<VideoFrameAllocator>>,
+) -> Vec<PlanarYCbCrImage> {
     let elements = unsafe { slice::from_raw_parts(elements, size) };
-    elements
-        .iter()
-        .map(|&img| -> PlanarYCbCrImage {
-            PlanarYCbCrImage {
-                picture: Region {
-                    x: img.mPicX,
-                    y: img.mPicY,
-                    width: img.mPicWidth,
-                    height: img.mPicHeight,
-                },
-                time_stamp: TimeStamp(img.mTimeStamp),
-                frame_id: img.mFrameID,
-                gecko_image: img,
-            }
-        })
-        .collect::<Vec<PlanarYCbCrImage>>()
+    let mut frames = vec![];
+    let video_frame_allocator = video_frame_allocator.lock().unwrap();
```

Comment:
Nit: perhaps use try_unwrap() here?

Reply:
I couldn't get this to work, looks like try_unwrap is on Rc only?